protocols/v2/codec-sv2/src/decoder.rs

```diff
@@ -1,3 +1,13 @@
+//! # Decoder
+//!
+//! Provides functionality for decoding Stratum V2 messages, including support for
+//! the Noise protocol for secure communication.
+//!
+//! ## Usage
+//!
+//! This module is designed to be used to decode incoming Sv2 messages, with optional Noise
+//! protocol encryption support for secure communication.
```

Comment:
I am not sure the mention of `for secure communication` is relevant here or in L3, IIUC the codec does not care or know anything about secure/not secure. it is just able to decode/encode sv2 messages with or without noise. In regards to what noise is, I think that that should be part of `noise-sv2`

Reply:
I made a lot of changes to the doc cmts. Give it another look and let me know what you think.